community/louvain_common.go

```diff
@@ -35,11 +36,194 @@ func Q(g graph.Graph, communities [][]graph.Node, resolution float64) float64 {
 	}
 }
 
+// ReducedGraph is a modularised graph.
+type ReducedGraph interface {
+	graph.Graph
+
+	// Communities returns the community memberships
+	// of the nodes in the graph used to generate
+	// the reduced graph.
+	Communities() [][]graph.Node
+
+	// Structure returns the community structure of
+	// the current level of the module clustering.
+	// The first index of the returned value
```

Comment:
I did try to reconcile the implementation of Communities with the comment of Structure but somehow it did not fit in. I am very slow today, could you show me with a code snippet the correspondence between the first index of the returned value and the expanded graph?